Wrappers/Python/cil/optimisation/algorithms/FISTA.py

```diff
@@ -69,51 +140,187 @@ def __init__(self, initial=None, f=None, g=ZeroFunction(), **kwargs):
         self.y.axpby(((self.t_old-1)/self.t), 1, self.x, out=self.y)
-        
+
         self.x_old.fill(self.x)
 
-        
+
     def update_objective(self):
-        self.loss.append( self.f(self.x) + self.g(self.x) )    
-    
+        """ Updates the objective
+
+        .. math:: f(x) + g(x)
+
+        """
+        self.loss.append( self.f(self.x) + self.g(self.x) )
+
+class ISTA(FISTA):
+
+    r"""Iterative Shrinkage-Thresholding Algorithm, see :cite:`BeckTeboulle_b`, :cite:`BeckTeboulle_a`.
+
+    Iterative Shrinkage-Thresholding Algorithm (ISTA)
+
+    .. math:: x^{k+1} = \mathrm{prox}_{\alpha^{k} g}(x^{k} - \alpha^{k}\nabla f(x^{k}))
+
+    is used to solve
+
+    .. math:: \min_{x} f(x) + g(x)
+
+    where :math:`f` is differentiable, :math:`g` has a *simple* proximal operator and :math:`\alpha^{k}`
+    is the :code:`step_size` per iteration.
+
+    Note
+    ----
+
+    For a constant step size, i.e., :math:`a^{k}=a` for :math:`k\geq1`, convergence of ISTA
+    is guaranteed if
+
+    .. math:: \alpha\in(0, \frac{2}{L}),
+
+    where :math:`L` is the Lipschitz constant of :math:`f`, see :cite:`CombettesValerie`.
+
+    Parameters
+    ----------
+
+    initial : DataContainer
+              Initial guess of ISTA.
+    f : Function
+        Differentiable function
+    g : Function
+        Convex function with *simple* proximal operator
+    step_size : positive :obj:`float`, default = None
+                Step size for the gradient step of ISTA.
+                The default :code:`step_size` is :math:`\frac{0.99 * 2}{L}.`
+    kwargs: Keyword arguments
+        Arguments from the base class :class:`.Algorithm`.
+
+
+
+    Examples
+    --------
+
+    .. math:: \underset{x}{\mathrm{argmin}}\|A x - b\|^{2}_{2}
+
+
+    >>> from cil.optimisation.algorithms import ISTA
+    >>> import numpy as np
+    >>> from cil.framework import VectorData
+    >>> from cil.optimisation.operators import MatrixOperator
+    >>> from cil.optimisation.functions import LeastSquares, ZeroFunction
+    >>> np.random.seed(10)
+    >>> n, m = 50, 500
+    >>> A = np.random.uniform(0,1, (m, n)).astype('float32') # (numpy array)
+    >>> b = (A.dot(np.random.randn(n)) + 0.1*np.random.randn(m)).astype('float32') # (numpy vector)
+    >>> Aop = MatrixOperator(A) # (CIL operator)
+    >>> bop = VectorData(b) # (CIL VectorData)
+    >>> f = LeastSquares(Aop, b=bop, c=0.5)
+    >>> g = ZeroFunction()
+    >>> ig = Aop.domain
+    >>> ista = ISTA(initial = ig.allocate(), f = f, g = g, max_iteration=10)
+    >>> ista.run()
+
+
+    See also
+    --------
+
+    :class:`.FISTA`
+    :class:`.GD`
+
+
+    """
+
+    @property
+    def convergence_criterion(self):
+        return self.step_size > 0.99*2.0/self.f.L
+
+    def __init__(self, initial, f, g, step_size = None, **kwargs):
+
+        super(ISTA, self).__init__(initial=initial, f=f, g=g, step_size=step_size, **kwargs)
+
+    def _set_step_size(self, step_size):
```

Comment:
see above comments in `FISTA`. This should be renamed to `set_step_size`

Reply:
Fixed.